File: 0242-valid-anagram/0242-valid-anagram.go
func isAnagram(s string, t string) bool {
    count := make(map[rune]int, len(s))
    for _, l := range s {
        count[l]++
    }
    for _, l := range t {
        count[l]--
    }
    for _, v := range count {
        if v != 0 {
            return false
        }
    }
    return true
}

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    let count = {}
    for (let c of s) {
        count[c] = (count[c] || 0) + 1
    }
    
    for (let c of t) {
        if (count[c] === undefined || count[c] == 0) {
            return false
        }
        count[c] = count[c] - 1
    }
    return s.length == t.length
};